syft/pkg/cataloger/java/graalvm_native_image_cataloger.go

```diff
@@ -135,58 +145,71 @@ func getPackage(component nativeImageComponent) pkg.Package {
 				GroupID: component.Group,
 			},
 		},
-		CPEs: cpes,
+		CPEs: getCPEs(component),
+		PURL: component.PackageURL,
 	}
+	if component.BOMRef != "" {
+		// explicitly setting the id is required for the "dependencies" field to be populated
+		p.OverrideID(artifact.ID(component.BOMRef))
+	}
+	return p
 }
 
 // decompressSbom returns the packages given within a native image executable's SBOM.
-func decompressSbom(dataBuf []byte, sbomStart uint64, lengthStart uint64) ([]pkg.Package, error) {
+func decompressSbom(dataBuf []byte, sbomStart uint64, lengthStart uint64) ([]pkg.Package, []artifact.Relationship, error) {
 	var pkgs []pkg.Package
 
 	lengthEnd := lengthStart + 8
 	bufLen := len(dataBuf)
 	if lengthEnd > uint64(bufLen) {
-		return nil, errors.New("the 'sbom_length' symbol overflows the binary")
+		return nil, nil, errors.New("the 'sbom_length' symbol overflows the binary")
 	}
 
 	length := dataBuf[lengthStart:lengthEnd]
 	p := bytes.NewBuffer(length)
 	var storedLength uint64
 	err := binary.Read(p, binary.LittleEndian, &storedLength)
 	if err != nil {
-		return nil, fmt.Errorf("could not read from binary file: %w", err)
+		return nil, nil, fmt.Errorf("could not read from binary file: %w", err)
 	}
 
 	log.WithFields("len", storedLength).Trace("found java native-image SBOM")
 	sbomEnd := sbomStart + storedLength
 	if sbomEnd > uint64(bufLen) {
-		return nil, errors.New("the sbom symbol overflows the binary")
+		return nil, nil, errors.New("the sbom symbol overflows the binary")
 	}
 
 	sbomCompressed := dataBuf[sbomStart:sbomEnd]
 	p = bytes.NewBuffer(sbomCompressed)
 	gzreader, err := gzip.NewReader(p)
 	if err != nil {
-		return nil, fmt.Errorf("could not decompress the java native-image SBOM: %w", err)
+		return nil, nil, fmt.Errorf("could not decompress the java native-image SBOM: %w", err)
 	}
 
 	output, err := io.ReadAll(gzreader)
 	if err != nil {
-		return nil, fmt.Errorf("could not read the java native-image SBOM: %w", err)
+		return nil, nil, fmt.Errorf("could not read the java native-image SBOM: %w", err)
 	}
 
-	var sbomContent nativeImageCycloneDX
-	err = json.Unmarshal(output, &sbomContent)
-	if err != nil {
-		return nil, fmt.Errorf("could not unmarshal the java native-image SBOM: %w", err)
+	var sbom cyclonedx.BOM
+	if err := json.Unmarshal(output, &sbom); err != nil {
```

Comment:
It might be best to use a [cyclondex.BOMDecoder](https://pkg.go.dev/github.com/CycloneDX/cyclonedx-go#BOMDecoder) or even [format.Decode](https://github.com/anchore/syft/blob/main/syft/format/decoders.go#L36C6-L36C75) -- if we used the `format.Decode`, it _absolutely_ should handle the dependency mapping to relationships for CycloneDX so we wouldn't have duplicated code here or even have to know anything about cyclonedx specifics

Reply:
Thanks for your feedback. I updated the code to call
```
sbom, _, _, err := cyclonedxjson.NewFormatDecoder().Decode(gzreader)
```
and simply return the packages and relationships. This effectively reuses the existing infrastructure ✅ 

This should resolve your concerns. Please have a look.

Reply:
> This reuses the existing infrastructure in an effective way

Perfect! Thanks much; this looks great!